src/numericalnim/utils.nim

```diff
@@ -262,104 +264,173 @@ proc calcError*[T](y_true, y: T): float {.inline.} = mean_squared_error(y_true,
-    let h11 = t ^ 3 - t ^ 2
-    result = h00 * y1 + h10 * (x2 - x1) * dy1 + h01 * y2 + h11 * (x2 - x1) * dy2
+  let t = (x - x1)/(x2 - x1)
+  let h00 = (1.0 + 2.0 * t) * (1.0 - t) ^ 2
+  let h10 = t * (1.0 - t) ^ 2
+  let h01 = t ^ 2 * (3.0 - 2.0 * t)
+  let h11 = t ^ 3 - t ^ 2
+  result = h00 * y1 + h10 * (x2 - x1) * dy1 + h01 * y2 + h11 * (x2 - x1) * dy2
 
 
 proc hermiteInterpolate*[T](x: openArray[float], t: openArray[float],
-                            y, dy: openArray[T]): seq[T] {.inline.} =
-    # loop over each interval and check if x is in there, if x is sorted
-    var xIndex = 0
-    if isSorted(x):
+              y, dy: openArray[T]): seq[T] {.inline.} =
+  # loop over each interval and check if x is in there, if x is sorted
+  var xIndex = 0
+  if isSorted(x):
+    for i in 0 .. t.high - 1:
+      while t[i] <= x[xIndex] and x[xIndex] < t[i+1]:
+        result.add(hermiteSpline(x[xIndex], t[i], t[i+1], y[i], y[i+1], dy[i], dy[i+1]))
+        xIndex += 1
+        if x.high < xIndex:
+          break
+      if x.high < xIndex:
+        break
+    if x[x.high] == t[t.high]:
+      result.add(y[y.high])
+  # loop over each x and then loop over each interval, if x not sorted
+  else:
+    for a in x:
+      block forblock:
         for i in 0 .. t.high - 1:
-            while t[i] <= x[xIndex] and x[xIndex] < t[i+1]:
-                result.add(hermiteSpline(x[xIndex], t[i], t[i+1], y[i], y[i+1], dy[i], dy[i+1]))
-                xIndex += 1
-                if x.high < xIndex:
-                    break
-            if x.high < xIndex:
-                break
-        if x[x.high] == t[t.high]:
-            result.add(y[y.high])
-    # loop over each x and then loop over each interval, if x not sorted
-    else:
-        for a in x:
-            block forblock:
-                for i in 0 .. t.high - 1:
-                    if t[i] <= a and a < t[i+1]:
-                        result.add(hermiteSpline(a, t[i], t[i+1], y[i], y[i+1], dy[i], dy[i+1]))
-                        break forblock
-                if a == t[t.high]:
-                    result.add(y[y.high])
-                else:
-                    raise newException(ValueError, &"{a} not in interval {min(t)} - {max(t)}")
-
+          if t[i] <= a and a < t[i+1]:
+            result.add(hermiteSpline(a, t[i], t[i+1], y[i], y[i+1], dy[i], dy[i+1]))
+            break forblock
+        if a == t[t.high]:
+          result.add(y[y.high])
+        else:
+          raise newException(ValueError, &"{a} not in interval {min(t)} - {max(t)}")
 
 
+proc delete*[T](s: var seq[T], idx: seq[int]) =
+  ## Deletes the elements of seq s at indices idx.
+  ## idx must not contain duplicates!
 
+  # sort indices in descending order so we can just delete them one by one without having to adjust the other indices
+  let indices = idx.sorted(Descending)
+  for i in indices:
+    s.delete(i)
+  
 
+proc getIndiceTable*[T](x: openArray[T]): Table[T, seq[int]] =
```

Comment:
Indice ↦ Index? Or Indices? :smile:

Reply:
Ugh I don't like using the plural of index but when I do I do it wrong 🤣 I'll go with `getIndexTable` for simplicity :)